src/realm/object-store/sync/impl/sync_metadata.cpp

```diff
@@ -182,7 +183,20 @@ SyncMetadataManager::SyncMetadataManager(std::string path, bool should_encrypt,
         }
     };
 
-    SharedRealm realm = Realm::get_shared_realm(config);
+    m_metadata_config = std::move(config);
+    SharedRealm realm;
+    try {
+        realm = get_realm();
+    }
+    catch (const std::exception& e) {
+        if (!should_encrypt)
+            throw(e);
+
+        assert(file_manager.remove_metadata_realm());
```

Comment:
`assert(...)` compiles to nothing in release builds, so this only deletes the file in debug builds. It also shouldn't be an assertion anyway, and should just do `File::remove(m_metadata_config.path);` rather than coupling this to `SyncFileManager` (which also eliminates the need to recreate the directory).

Reply:
> assert(...) compiles to nothing in release builds, so this only deletes the file in debug builds.
Thanks! I think explains the release failure in the ci, too.

Reply:
> It also shouldn't be an assertion anyway, and should just do File::remove(m_metadata_config.path);

Does this mean that keeping the lock and management files around is okay?

Reply:
There's no persistent data in anything but the realm file itself so there's no need to delete them.